### PR TITLE
Simplify GradleTestKit usage.

### PR DESCRIPTION
**Pull request by cdancy**
@bmuschko if you get a chance can you look things over? I took examples from what you've done on your other projects and applied it here to simplify test kit usage. It works as expected just wondering if there is anything that can be done better as I assume you're much closer to the internals and understandings of how this works than I am.

### CODE REVIEW

File: src/functTest/groovy/com/bmuschko/gradle/docker/AbstractFunctionalTest.groovy [hunk 1]
```diff
@@ -33,26 +33,21 @@ abstract class AbstractFunctionalTest extends Specification {
...
-
-        def pluginClasspath = pluginClasspathResource.readLines()
-                .collect { it.replace('\\', '\\\\') } // escape backslashes in Windows paths
-                .collect { "'$it'" }
-                .join(", ")
-
-        // Add the logic under test to the test build
+                
+        def pluginClasspath = GradleRunner.create().withPluginClasspath().getPluginClasspath()
+            .collect { it.path.replace('\\', '\\\\') } // escape backslashes in Windows paths
```
Comment (bmuschko) [inline src/functTest/groovy/com/bmuschko/gradle/docker/AbstractFunctionalTest.groovy]: Lines 38-40 are not needed anymore.

File: src/functTest/groovy/com/bmuschko/gradle/docker/AbstractFunctionalTest.groovy [hunk 2]
```diff
@@ -33,26 +33,21 @@ abstract class AbstractFunctionalTest extends Specification {
...
+         // Add the logic under test to the test build
         buildFile << """
             buildscript {
                 dependencies {
                     classpath files($pluginClasspath)
                 }
             }
         """
-
+
```
Comment (bmuschko) [inline src/functTest/groovy/com/bmuschko/gradle/docker/AbstractFunctionalTest.groovy]: Extra new line?

**Comment by bmuschko** [comment]
Thought I had changed it in this plugin as well. Thanks for picking it up.

File: src/functTest/groovy/com/bmuschko/gradle/docker/AbstractFunctionalTest.groovy (hunk 2 above)
Comment (cdancy) [inline src/functTest/groovy/com/bmuschko/gradle/docker/AbstractFunctionalTest.groovy]: Looks like a few tabs were inserted. Weird. I'll clean that up.

File: src/functTest/groovy/com/bmuschko/gradle/docker/AbstractFunctionalTest.groovy (hunk 1 above)
Comment (cdancy) [inline src/functTest/groovy/com/bmuschko/gradle/docker/AbstractFunctionalTest.groovy]: @bmuschko removing them causes functional tests to fail due to inability to find all classes with plugin. I have to imagine this is due to the way we are resolving the dependencies internally. Still cleaner than what we had before however.

Comment (bmuschko) [inline src/functTest/groovy/com/bmuschko/gradle/docker/AbstractFunctionalTest.groovy]: Oh, I see what's wrong here. The buildscript shouldn't use the `buildscript` notation anymore. Instead it should use the plugin DSL. See this [commit](https://github.com/bmuschko/gradle-tomcat-plugin/commit/bacc2dc9ad5695f67b4b9f5b4e6509c05c3f8def) for more info.

Comment (cdancy) [inline src/functTest/groovy/com/bmuschko/gradle/docker/AbstractFunctionalTest.groovy]: @bmuschko that failed as well with the below error complaining about version. The examples you cited don't use a version so I suspect things should be wired in such a way as to pick the very thing that was just built?

* What went wrong:
Plugin [id: 'com.bmuschko.gradle.docker.DockerRemoteApiPlugin'] was not found in any of the following sources:

- Gradle Core Plugins (plugin is not in 'org.gradle' namespace)
- Gradle TestKit (classpath: /home/cdancy/github-projects/gradle-docker-plugin/build/classes/main:/home/cdancy/github-projects/gradle-docker-plugin/build/resources/main)
- Gradle Central Plugin Repository (plugin dependency must include a version number for this source)

Relevant code snippet:

    def setup() {
        projectDir = temporaryFolder.root
        buildFile = temporaryFolder.newFile('build.gradle')

        buildFile << """
            plugins {
                id 'com.bmuschko.gradle.docker.DockerRemoteApiPlugin'
            }

			apply plugin: 'java'
			
            repositories {
                mavenCentral()
            }
        """

        setupDockerServerUrl()
        setupDockerCertPath()
        setupDockerPrivateRegistryUrl()
    }

Comment (bmuschko) [inline src/functTest/groovy/com/bmuschko/gradle/docker/AbstractFunctionalTest.groovy]: @cdancy You'll have to use the plugin identifier as String and not the fully-qualified class name.

    plugins {
         id 'com.bmuschko.docker-remote-api'
    }

Comment (cdancy) [inline src/functTest/groovy/com/bmuschko/gradle/docker/AbstractFunctionalTest.groovy]: @bmuschko #facepalm. Can't believe I missed that. Works as expected. Cleaning some things up and will check in the change momentarily.

**Pull request merged**